spec/models/iiif_presentation_spec.rb

```diff
@@ -138,4 +138,70 @@ def stub_info
       expect(iiif_presentation.manifest.to_json(pretty: true)).to include "Strawberry Thief fabric, made by Morris and Company "
     end
   end
+
+  describe 'iiif presentation validations' do
+    context 'manifests' do
+      manifest = IIIF::Presentation::Manifest.new
+
+      it 'raises an exception if the manifest does not have an id' do
+        manifest.label = 'Book 1'
+        expect { manifest.validate }.to raise_error IIIF::Presentation::MissingRequiredKeyError
+      end
+
+      it 'raises an exception if the manifest does not have an label' do
+        manifest['id'] = 'http://www.example.org/iiif/book1/manifest'
+        expect { manifest.validate }.to raise_error IIIF::Presentation::MissingRequiredKeyError
+      end
+
+      it 'raises an exception if the manifest does not have a type' do
+        manifest.delete('@type')
+        manifest.label = 'Book 1'
+        manifest['id'] = 'http://www.example.org/iiif/book1/manifest'
+        expect { manifest.validate }.to raise_error IIIF::Presentation::MissingRequiredKeyError
+      end
+    end
+
+    it 'raises an exception if the sequence is not an array' do
+      expect { (iiif_presentation.manifest.sequences = 'quux') }.to raise_error('sequences must be an Array.')
+    end
+
+    it 'raises an exception if the canvas dimensions are not present' do
+      canvas = IIIF::Presentation::Canvas.new('@id' => 'http://example.com/canvas')
+      expect { canvas.to_json }.to raise_error('A(n) width is required for each IIIF::Presentation::Canvas')
+      canvas.width = 191
+      expect { canvas.to_json }.to raise_error('A(n) height is required for each IIIF::Presentation::Canvas')
+    end
+
+    it 'raises an exception if the images service_id is not included' do
+      expect { IIIF::Presentation::ImageResource.create_image_api_image_resource }.to raise_error('key not found: :service_id')
+    end
+
+    # #TODO: Fix this test. Not currently working as at should since we are stubbing requests
+    xit 'raises an exception if the images service_id is not valid' do
```

Comment:
Do you want to get this passing before merging? I could help you with how to stub it, I think.